### ISSUE TITLE
Signature is base64-encoded with Qt Core v5.15.0

### ISSUE THREAD

**Issue by andres-liiver** [open-eid/chrome-token-signing#168]
When compiling the Linux version with Qt Core v5.15.0, the signature hex string is base64-encoded. This breaks [hwcrypto.js](https://github.com/hwcrypto/hwcrypto.js) which expects a hex string, not base64.

**Comment by neonsea** [comment]
Also happens on my machine. Worked fine before qt5-core upgrade.

**Comment by neonsea** [comment]
![Pic](https://arse.ee/JQxBiQm5.png)

**Comment by saunaants** [comment]
Conversion from QByteArray to QJsonValue is not supported
https://github.com/open-eid/chrome-token-signing/blob/c0a4ae1021d29d55faf9a07b67559202b33450a1/host-linux/Signer.h#L164

https://doc.qt.io/qt-5/qjsonvalue.html#fromVariant

**Comment by metsma** [comment]
I don't understand where it gets base64 convertede.
Version 15.5 ? you mean 5.15.0 ?

**Comment by saunaants** [comment]
qt 5.15 reimplemented JSON on top of CBOR https://code.qt.io/cgit/qt/qtbase.git/commit/?id=35adb74ddd915831789f0175423660f8e898942e

in CBOR arrays get base64 encoded https://doc.qt.io/qt-5/qcborvalue.html#toJsonValue

**Comment by andres-liiver** [comment]
> I don't understand where it gets base64 convertede.
> Version 15.5 ? you mean 5.15.0 ?

Oops, 5.15.0, yes. Not sure how that got messed up. :laughing:

**Comment by metsma** [comment]
https://bugreports.qt.io/browse/QTBUG-84739

**Comment by yltsparri** [comment]
Quick fix is to convert hex value from QByteArray to QString.
My quick patch that seems to work:

```patch
--- a/host-linux/Signer.h
+++ b/host-linux/Signer.h
@@ -159,9 +159,9 @@ public:
     }
 
 private:
-    static QByteArray toHex(const std::vector<unsigned char> &data)
+    static QString toHex(const std::vector<unsigned char> &data)
     {
-        return QByteArray::fromRawData((const char*)data.data(), int(data.size())).toHex();
+        return QString(QByteArray::fromRawData((const char*)data.data(), int(data.size())).toHex());
     }
 
     static std::vector<unsigned char> fromHex(const QString &data)
```

**Comment by metsma** [comment]
They fixed/reverted the behavior in 5.15.1
https://github.com/qt/qtbase/blob/5.15.1/dist/changes-5.15.1#L27-L32
